src/translation/Translation.ts

```diff
@@ -0,0 +1,45 @@
+import trJson from './tr.json'
+
+export class Translation {
+    data: any
+    fallback: any
+
+    constructor(data: any, fallback: any) {
+        if (!data) throw Error('Translation is undefined')
+        if (!fallback) throw Error('Translation fallback is undefined')
+        this.data = data
+        this.fallback = fallback
+    }
+
+    tr(key: string, parameters?: string[]): string {
+        var str: string = this.data[key]
+        if (!str) {
+            str = this.fallback[key]
+            if (!str) return key
+        }
+        if (parameters)
+            for (let i = 0; i < parameters.length; i++) {
+                str = str.replace('%' + (i + 1) + '$s', parameters[i])
+            }
+        return str || ''
+    }
+}
+
+let translation: Translation
+export function setTranslation(lang: string, overwrite = false): Translation {
+    if (translation && !overwrite) throw new Error('translation already initialized')
+
+    let json = (trJson as any) as Record<string, any>
```

Comment:
I would think that you only need to cast once ``` const json = trJson as Record<string, any>```

Reply:
I guess any as record value is fine here, because we don't really care in this scope

Reply:
Have changed it to be more precise.